tasks/main.yml

```diff
@@ -58,12 +58,15 @@
   with_dict: riak_conf
 
 - name: Tag riak.conf as ansible managed
-  lineinfile: "dest=/etc/riak/riak.conf regexp='^## Ansible Managed.+$'  line='## Ansible Managed\n' insertbefore='BOF'"
+  lineinfile: dest=/etc/riak/riak.conf regexp='^## Ansible Managed.*$'  line='## Ansible Managed' insertbefore='BOF'
   tags: configfiles
 
-- name: Make certain Riak is set to start on boot
+- name: Make certain Riak is started and set to start on boot
   service: name=riak enabled=yes state=started
 
+- name: Wait for Riak to start up before continuing
+  wait_for: delay=5 timeout=30 host={{ inventory_hostname }} port={{ riak_pb_port }} state=started
```

Comment:
Thanks for your feedback, I identified the issue with an invalid node name this past week as well. I am going to bring up that issue with the core engineering team at our engineering all hands in a week.